File: backend/internal/core/domain/subject.go
package domain

import "gorm.io/gorm"

type Subject struct {
	gorm.Model
	Code        string `gorm:"uniqueIndex;not null;type:varchar(20)" json:"code" example:"MAT101"` // รหัสวิชา (ห้ามซ้ำ)
	Name        string `gorm:"not null;type:varchar(100)" json:"name" example:"Mathematics"`       // ชื่อวิชา
	Description string `gorm:"type:text" json:"description" example:"วิชาคณิตศาสตร์พื้นฐาน"`           // คำอธิบาย
	IsActive    bool   `gorm:"default:true" json:"is_active" example:"true"`                       // เปิดใช้งานหรือไม่
}

File: frontend/src/types/subject.ts
export interface Subject {
  ID: number;
  code: string;
  name: string;
  description: string;
  is_active: boolean;
  CreatedAt: string;
  UpdatedAt: string;
}

// ข้อมูลสำหรับตอนสร้างหรือแก้ไข (ไม่ต้องส่ง ID ไป)
export interface SubjectFormInput {
  code: string;
  name: string;
  description?: string;
}